apollo-gradle-plugin/src/main/groovy/com/apollostack/android/gradle/ApolloIRgenTask.groovy

```diff
@@ -0,0 +1,83 @@
+package com.apollostack.android.gradle
+
+import com.google.common.collect.Sets
+import com.moowork.gradle.node.task.NodeTask
+import org.gradle.api.GradleException
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.OutputDirectory
+import org.gradle.api.tasks.util.PatternSet
+
+public class ApolloIRgenTask extends NodeTask {
+  private static final String APOLLO_CODEGEN = "node_modules/apollo-codegen/lib/cli.js"
+  private static final String DEFAULT_OUTPUT_DIR = "src/main/graphql"
+
+  static final String NAME = "generate%sApolloIR"
+  List<ApolloExtension> config
+  String variant
+
+  @OutputDirectory
```

Comment:
@sav007 updated the task to be incremental. Gradle would check the input files and output directory. If none of the inputs or outputs have changed, the task is skipped.

```
Executing external task 'generateDebugApolloIR'...
Configuration on demand is an incubating feature.
Incremental java compilation is an incubating feature.
:apollo-sample:nodeSetup UP-TO-DATE
:apollo-sample:installApolloCodegen UP-TO-DATE
:apollo-sample:generateDebugApolloIR UP-TO-DATE
```

The `ApolloClassgenTask` will be made incremental too.